src/backend/mysql.rs

```diff
@@ -0,0 +1,110 @@
+//! MySQL implementation of a generator
+//!
+//! This module generates strings that are specific to MySQL
+//! databases. They should be thoroughly tested via unit testing
+
+use super::SqlGenerator;
+use types::{BaseType, Type};
+
+/// MySQL generator backend
+pub struct MySql;
+impl SqlGenerator for MySql {
+    fn create_table(name: &str) -> String {
+        format!("CREATE TABLE {}", name)
+    }
+
+    fn create_table_if_not_exists(name: &str) -> String {
+        format!("CREATE TABLE {} IF NOT EXISTS", name)
+    }
+
+    fn drop_table(name: &str) -> String {
+        format!("DROP TABLE {}", name)
+    }
+
+    fn drop_table_if_exists(name: &str) -> String {
+        format!("DROP TABLE {} IF EXISTS", name)
+    }
+
+    fn rename_table(old: &str, new: &str) -> String {
+        format!("RENAME TABLE \"{}\" TO \"{}\"", old, new)
+    }
+
+    fn alter_table(name: &str) -> String {
+        format!("ALTER TABLE \"{}\"", name)
+    }
+
+    fn add_column(ex: bool, name: &str, tt: &Type) -> String {
+        let bt: BaseType = tt.get_inner();
+        use self::BaseType::*;
+
+        #[cfg_attr(rustfmt, rustfmt_skip)] /* This shouldn't be formatted. It's too long */
+        format!(
+            "{}{}{}",
+            match bt {
+                Text => format!("{}{} {}", MySql::prefix(ex), name, MySql::print_type(bt)),
+                Varchar(_) => format!("{}{} {}", MySql::prefix(ex), name, MySql::print_type(bt)),
+                Primary => format!("{}{} {}", MySql::prefix(ex), name, MySql::print_type(bt)),
+                Integer => format!("{}{} {}", MySql::prefix(ex), name, MySql::print_type(bt)),
+                Float => format!("{}{} {}", MySql::prefix(ex), name, MySql::print_type(bt)),
+                Double => format!("{}{} {}", MySql::prefix(ex), name, MySql::print_type(bt)),
+                UUID => unimplemented!(),
```

Comment:
According to the MySQL 8.0 docs there are supported UUID fields. This is definitely something that `barrel` should reflect.

Unrelated: this also makes me wonder about how to document the _vast_ amounts of implementation specific features across multiple versions as well :sweat_smile: